src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Flows/AcquireTokenHandlerBase.cs

```diff
@@ -220,12 +223,27 @@ public async Task<AuthenticationResult> RunAsync()
             catch (Exception ex)
             {
                 RequestContext.Logger.ErrorPii(ex);
-                if (_client != null && _client.Resiliency && extendedLifetimeResultEx != null)
+
+                if (_client == null)
+                {
+                    _client = new OAuthClient();
+                }
```

Comment:
see below, you have an opportunity to replace this with a flag instead of newing up this object